unimol/unimol/models/unimol.py

```diff
@@ -229,14 +229,18 @@ def get_dist_features(dist, et):
             if self.args.masked_coord_loss > 0:
                 coords_emb = src_coord
                 if padding_mask is not None:
-                    atom_num = (torch.sum(1 - padding_mask.type_as(x), dim=1) - 1).view(
+                    atom_num = torch.sum(1 - padding_mask.type_as(x), dim=1).view(
```

Comment:
Changes here may not be necessary. EOS is not used

Reply:
Yes.I don't think EOS is relevant here. 
The only thing relevant is that you sum after with the following mask 
`pair_coords_mask = (1 - padding_mask.float()).unsqueeze(-1) * (1 - padding_mask.float()).unsqueeze(1)`
you can keep the `-1` in `atom_num` because indeed for i=j, the difference of coordinates is zero.
![image](https://github.com/dptech-corp/Uni-Mol/assets/31697743/81025ab2-525c-42e9-a256-bba12333db03)

Reply:
Agree with your point. Then the `-1` in the `else`  also needs to be removed.

Reply:
Done ! Thanks.